app/assets/javascripts/discourse/app/lib/do-not-disturb.js

```diff
@@ -0,0 +1,39 @@
+import DiscourseURL from "discourse/lib/url";
+import { Promise } from "rsvp";
+import { ajax } from "discourse/lib/ajax";
+
+export function publishDoNotDisturbOnFor(user, duration) {
+  return new Promise(function (resolve, reject) {
+    ajax({
+      url: "/do-not-disturb.json",
+      type: "POST",
+      data: { duration: duration },
+    })
+      .then((response) => {
+        updateDoNotDisturbStatus(user, response.ends_at);
+        return resolve();
+      })
+      .catch((error) => reject(error));
+  });
+}
+
+export function publishDoNotDisturbOffFor(user) {
+  return new Promise(function (resolve, reject) {
+    ajax({
+      url: "/do-not-disturb.json",
+      type: "DELETE",
+    })
+      .then(() => {
+        updateDoNotDisturbStatus(user, null);
+        return resolve();
+      })
+      .catch(reject);
+  });
+}
+export function updateDoNotDisturbStatus(user, ends_at) {
```

Comment:
I suggest moving this to the `user` model, so you can call it with `this.currentUser.updateDoNotDisturbStatus(endsAt)`